README.md

```diff
@@ -3,7 +3,71 @@
 [![Swift 5.2](https://img.shields.io/badge/Swift-5.2-ED523F.svg?style=flat)](https://swift.org/download/)
 [![CI](https://github.com/slashmo/gsoc-swift-baggage-context/workflows/CI/badge.svg)](https://github.com/slashmo/gsoc-swift-baggage-context/actions?query=workflow%3ACI)
 
-`BaggageContext` originated in [this repo](https://github.com/slashmo/gsoc-swift-tracing), but had to be factored out so that we can depend on it in our [NIO fork](https://github.com/slashmo/swift-nio).
+`BaggageContext` is a minimal (zero-dependency) "context" library meant to "carry" baggage (metadata) for cross-cutting tools such as tracers.
+It is purposefully not tied to any specific use-case (in the spirit of the Tracing Plane paper's BaggageContext), however does not implement its own serialization scheme (today), however should enable vast majority of use cases cross-cutting tools need to support.
+
+See https://github.com/slashmo/gsoc-swift-tracing for actual instrument types and implementations which can be used to deploy various cross-cutting instruments all reusing the same baggage type.
+
+## Context-Passing Guidelines
+
+In order for context-passing to feel consistent and Swifty among all server-side (and not only) libraries and frameworks 
+aiming to adopt `BaggageContext` (or any of its uses, such as Distributed Tracing), we suggest the following set of guidelines:
+
+### Argument naming/positioning
+
+In order to propagate baggage through function calls (and asynchronous-boundaries it may often be necessary to pass it explicitly (unless wrapper APIs are provided which handle the propagation automatically). 
+
+When passing baggage context explicitly we strongly suggest sticking to the following style guideline:
+
+- Assuming the general parameter ordering of Swift function is as follows (except DSL exceptions):
+  1. Required non-function parameters (e.g. `(url: String)`),
+  2. Defaulted non-function parameters (e.g. `(mode: Mode = .default)`),
+  3. Required function parameters, including required trailing closures (e.g. `(onNext elementHandler: (Value) -> ())`),
+  4. Defaulted function parameters, including optional trailing closures (e.g. `(onComplete completionHandler: (Reason) -> ())`).
```

Comment:
```suggestion
  4. Defaulted function parameters, including optional trailing closures (e.g. `(onComplete completionHandler: (Reason) -> ()) = { _ in }`).
```

To illustrate defaulted function parameters, we should either show a default closure or an optional closure.

Reply:
👍